jira/client.py

```diff
@@ -5391,7 +5391,7 @@ def update_sprint(
             payload["goal"] = goal
 
         url = self._get_url(f"sprint/{id}", base=self.AGILE_BASE_URL)
-        r = self._session.put(url, data=json.dumps(payload))
+        r = self._session.post(url, data=json.dumps(payload))
```

Comment:
```suggestion
        method = self._session.post if partial else self._session.put
        r = method(url, data=json.dumps(payload))
```

(with corresponding addition of `partial` to args